colcon_cmake/task/cmake/build.py

```diff
@@ -313,16 +314,30 @@ def _get_make_arguments(self):
 
         if CMAKE_EXECUTABLE is None:
             raise RuntimeError("Could not find 'cmake' executable")
-        cmd = [
-            CMAKE_EXECUTABLE, '--build', args.build_base,
-            '--target', 'install']
+        cmake_ver = await get_cmake_version()
+        cmd = [CMAKE_EXECUTABLE]
+        if cmake_ver[0] > 3 or cmake_ver[0] == 3 and cmake_ver[1] >= 15:
+            # CMake 3.15+ supports invoking `cmake --install [--config]
+            # This only installs, whereas --build <dir> --target install will
+            # build (again) then install.
+            cmd += ['--install', args.build_base]
+        else:
+            # Fallback to building the install target.
+            cmd += ['--build', args.build_base, '--target', 'install']
+
         multi_configuration_generator = is_multi_configuration_generator(
             args.build_base, args.cmake_args)
         if multi_configuration_generator:
             cmd += ['--config', self._get_configuration(args)]
         else:
+            # Add make/ninja arguments for the number of job threads to use.
             job_args = self._get_make_arguments()
             if job_args:
+                # Add arguments for the number of job threads to use
+                # Arguably, _get_make_arguments() should be be configured to
+                # return arguments based on the current generator, not just for
+                # make (and ninja by coincidence). The '--' extension may be
+                # better returned from _get_make_arguments()
```

Comment:
Please remove those comments which have been removed from the other PR.

Reply:
Will do. Just catching up on various comments.